### ISSUE TITLE
LongParameterList ignoreAnnotated not working as expected

### ISSUE THREAD

**Issue by lasterketaDev** [detekt/detekt#3395]
## Issue

We set the LongParameterList option to ignore the following classes: 
-dagger.Component.Factory

But detekt flags this code:

```
    @Component.Factory
    interface Factory {
        fun create(
            @BindsInstance app: MyApp,
            component1: Component1,
            component2: Component2,
            component3: Component3,
            component4: Component4
        ): ApplicationComponent
    }
```
when it should not flag it, as it is in the ignore list.

## Context

It is a dagger component factory, so we need to pass multiple components.

## Your Environment

* Version of detekt used: 1.15.0
* Version of Gradle used (if applicable): 6.7.1

**Comment by BraisGabin** [comment]
The problem here is that we don't support type solving on this rule. So we don't support adding all the package. It works if you just add `Component.Factory`.

One more reason to generalize the way we allow to ignore some rules.
